src/kernels/normalizedkernel.jl

```diff
@@ -0,0 +1,81 @@
+"""
+    NormalizedKernel(k::Kernel)
+
+A normalized kernel derived from `k`.
+
+# Definition
+
+For inputs ``x, x'``, the normalized kernel ``\\widetilde{k}`` derived from
+kernel ``k`` is defined as
+```math
+\\widetilde{k}(x, x'; k) = \\frac{k(x, x')}{\\sqrt{k(x, x) k(x', x')}}.
+```
+"""
+struct NormalizedKernel{Tk<:Kernel} <: Kernel
+    kernel::Tk
+end
+
+@functor NormalizedKernel
+
+(κ::NormalizedKernel)(x, y) = κ.kernel(x, y) / sqrt(κ.kernel(x, x) * κ.kernel(y, y))
+
+function kernelmatrix(κ::NormalizedKernel, x::AbstractVector, y::AbstractVector)
+    return kernelmatrix(κ.kernel, x, y) ./
+           sqrt.(
+        kernelmatrix_diag(κ.kernel, x) .* permutedims(kernelmatrix_diag(κ.kernel, y))
+    )
+end
+
+function kernelmatrix(κ::NormalizedKernel, x::AbstractVector)
+    x_diag = kernelmatrix_diag(κ.kernel, x)
+    return kernelmatrix(κ.kernel, x) ./ sqrt.(x_diag .* permutedims(x_diag))
+end
+
+function kernelmatrix_diag(κ::NormalizedKernel, x::AbstractVector)
+    first_x = first(x)
+    return Fill(κ(first_x, first_x), length(x))
```

Comment:
Can't we replace `Fill` by `Ones` here?
```suggestion
    return Ones{typeof(κ(first_x, first_x))},(length(x))
```

Reply:
Or can some kernel return a negative value...

Reply:
Look like this fixes the AD, cheers :)

Reply:
I suppose k(x, x) could be zero as well?

Reply:
But then we're in trouble cause we would be dividing by 0 and you would probably have bigger problems generally.